### PR TITLE
Align request protocol with window.location

### PR DESCRIPTION
**Pull request by martletandco**
`window.location.protocol` includes the colon (`https:`) whereas `request.protocol` does not (`https`). As this adds an additional step in getting a consistent protocol value it, introduces a 'gotcha'.

This PR appends a colon to Request's location protocol so that

```
let location = get(this, 'fastboot.isFastBoot') ? get(this, 'fastboot.request') : window.location;
let protocol = get(location, 'protocol');
protocol = protocol.replace(':', '');
```

can just be

```
let location = get(this, 'fastboot.isFastBoot') ? get(this, 'fastboot.request') : window.location;
let protocol = get(location, 'protocol');
```

### CODE REVIEW

File: src/fastboot-request.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var FastBootHeaders = require('./fastboot-headers');
 function FastBootRequest(request, hostWhitelist) {
   this.hostWhitelist = hostWhitelist;
 
-  this.protocol = request.protocol;
+  this.protocol = request.protocol+':';
```
Comment (kratiahuja) [inline src/fastboot-request.js]: Spacing for `+` seems a bit odd here.

Comment (martletandco) [inline src/fastboot-request.js]: Is the preferred spacing `request.protocol + ':';`?

Comment (kratiahuja) [inline src/fastboot-request.js]: I assume so. Better to drop using string concatenation and instead use template liberals in ES6

Comment (martletandco) [inline src/fastboot-request.js]: Ok, good call. Have changed this to use a template literal.

**Comment by martletandco** [comment]
Just checking progress on this, are there any objections?

**Comment by kellyselden** [comment]
None from me

**Comment by kratiahuja** [comment]
@martletandco Could you resolve the conflicts? Since there are no strong objections, we should merge this. Also this would be good to get merged before FastBoot 1.0

**Comment by martletandco** [comment]
All up to date now. Didn't see that conflict sorry

**Comment by kratiahuja** [comment]
Thanks @martletandco

**Pull request merged**